### ISSUE TITLE
Can't get Pod table formatted right

### ISSUE THREAD

**Issue by zoffixznet** [rakudo/rakudo#1821]
I have this table in my code:

```perl6
=begin table

    Type    | Comments
    ========+=================================================
    Complex |
    Num     |
    FatRat  |
    MidRat  | Special infectiousness. See prose that follows. 
    Rat     |
    Int     |

=end table
```

Which errors out with this:
```
$ perl6 -c doc/Language/numerics.pod6                                                                                                                                                                   
===FATAL: Table 2 has a mixture of visible and invisible column-separator types.
    Type    | Comments
    ========+=================================================
    Complex |
    Num     |
    FatRat  |
    MidRat  | Special infectiousness. See prose that follows. 
    Rat     |
    Int     |

===end FATAL table 2 input
===SORRY!===
```

I've no idea what it's talking about since as far as I can tell I've only used visible separators for all the things I wanted to use separators for.

I changed the code to this:

```perl6
=begin table

    Type    | Comments                                        |
    ========+=================================================+
    Complex |                                                 |
    Num     |                                                 |
    FatRat  |                                                 |
    MidRat  | Special infectiousness. See prose that follows. |
    Rat     |                                                 |
    Int     |                                                 |

=end table
```

And now the fatal error is gone, but I get a warning instead:

```
$ perl6 -c doc/Language/numerics.pod6 
===WARNING: One or more tables evidence bad practice.
==          Set environment variable 'RAKUDO_POD_TABLE_DEBUG' for more details.
Syntax OK

$ RAKUDO_POD_TABLE_DEBUG=1 perl6 -c doc/Language/numerics.pod6
[...]
===WARNING: Table 2 has unneeded border vis col separators.
    Type    | Comments                                        |
    ========+=================================================+
    Complex |                                                 |
    Num     |                                                 |
    FatRat  |                                                 |
    MidRat  | Special infectiousness. See prose that follows. |
    Rat     |                                                 |
    Int     |                                                 |

===end WARNING table 2 input rows
[...]
```

First of all, why is drawing the full table a "bad practice"? What's bad about it? And second, what's the right way to draw such a table?

**Comment by zoffixznet** [comment]
> two adjacent embedded spaces in the filled second column?

You mean after `See prose that follows.`? No, there's just one space before the border and no spaces in the variant without the border.

**Comment by tbrowder** [comment]
On Sat, May 12, 2018 at 07:36 Zoffix Znet <notifications@github.com> wrote:

> two adjacent embedded spaces in the filled second column?
>
> You mean after See prose that follows.? No, there's just one space before
> the border and no spaces in the variant without the border.
>

See my update 2.  I’m pretty sure that will fix the immediate problem for
you.  I won’t be able to work on the code for a while, but please let me
know if the temp “fix” (removing the trailing pipes) works.

Best.

-Tom

**Comment by tbrowder** [comment]
Your “baby-sitting” comments are spot on.  Realize that when I finished that table pod work I was heavily invested in numerous edge cases and troublesome differences in vis and non-vis col sep tables, and I wanted to warn table authors about subtle traps.  I will change the warning approach when I fix your problem: remove all warnings as the default unless an appropriate environment variable is set.

**Comment by zoffixznet** [comment]
> UPDATE 2. I think I see the problem. Can you try REMOVING the trailing pipes in your original version (just leave the single pipe separating the two columns of text) and see if that solves the problem. I’m not able to compute at the moment but your test would help me later.

This still has the fatal error:
```perl6
=begin table

    Type    | Comments
    ========+=================================================
    Complex
    Num
    FatRat
    MidRat  | Special infectiousness. See prose that follows.
    Rat
    Int

=end table
```

```
===FATAL: Table 2 has a mixture of visible and invisible column-separator types.
    Type    | Comments
    ========+=================================================
    Complex
    Num
    FatRat
    MidRat  | Special infectiousness. See prose that follows.
    Rat
    Int

===end FATAL table 2 input
===SORRY!===
```

**Comment by tbrowder** [comment]
Closed with commit 35afcc9.
